apps/vmq_server/src/vmq_cluster_node.erl

```diff
@@ -174,21 +178,25 @@ handle_message({msg, CallerPid, Ref, Msg}, State) ->
             CallerPid ! {Ref, ok}
     end,
     NewState;
-handle_message({NetEv, _}, #state{reconnect_tref=TRef} = State)
-  when
-      NetEv == tcp_closed;
-      NetEv == tcp_error;
-      NetEv == ssl_closed;
-      NetEv == ssl_error ->
-    NewTRef =
-    case TRef of
-        undefined ->
-            reconnect_timer();
-        _ ->
-            %% we're already reconnecting
-            TRef
-    end,
-    State#state{reachable=false, reconnect_tref=NewTRef};
+handle_message({connect_async, AsyncPid, {ok, {Transport, Socket}}},
```

Comment:
I think you should rename `connect_async` to something like `connect_finished`. When I read the code the first time I thought `connect_async` was the message sent to start the connect process.